src/tabmat/constructor_util.py

```diff
@@ -0,0 +1,32 @@
+from typing import Tuple
```

Comment:
What is the motivation behind moving this out of `constructor.py`? If we keep it separate, then I would rather organize this in a separate package as:
- `constructor/constructor.py` (current `constructor.py`), 
- `constructor/utils.py` (current `constructor_util.py`), 
- `constructor/__init__.py` (exposes imports from `constructor/constructor.py`).

Then one can still write `from tabmat.constructor import from_pandas` etc. as before, but the constructor is more organized.

Reply:
The reason is that I ran into a circular import: I need `_split_sparse_dense` in both the `formula.py` and `constructor.py` modules, and we need stuff from `formula.py` in `constructor.py`. If this function is in the latter, then that makes it circular. Your solution sounds good, though. Other options include (if we want to keep the flat structure):

 - Move everything from `formula.py` into `constructor.py` (formulas are related to tabmat construction, anyways)
 - Move the `_split_sparse_dense` to `formula.py` and import in `constructor.py`. In that case we should probably call `formula.py` something else.